packages/core/test/specHelper.ts

```diff
@@ -133,16 +133,14 @@ class StubXhr {
   public status: number | undefined = undefined
   public readyState: number = XMLHttpRequest.UNSENT
   public onreadystatechange: () => void = noop
-
+  public listeners: { [k: string]: Array<() => void> } = {}
+  private isOnreadystatechangeAttributeCallFirst = true
```

Comment:
Maybe we could have something like this:

```
private _onreadystatechange: null | (() => void) = null

get onreadystatechange() {
  return this._onreadystatechange
}
set onreadystatechange(value) {
  if (!this._onreadystatechange) {
    this.addEventListener('readystatechange', () => this._onreadystatechange())
  }
  this._onreadystatechange = value
}
```

So we are sure that `readystatechange` listeners will be called in order, and we don't need to handle another state.